crossbeam-skiplist/src/base.rs

```diff
@@ -99,11 +99,8 @@ impl<K, V> Node<K, V> {
     /// with null pointers. However, the key and the value will be left uninitialized, and that is
     /// why this function is unsafe.
     unsafe fn alloc(height: usize, ref_count: usize) -> *mut Self {
-        // TODO(stjepang): Use the new alloc API instead of this hack once it becomes stable.
-        let cap = Self::size_in_u64s(height);
-        let mut v = Vec::<u64>::with_capacity(cap);
-        let ptr = v.as_mut_ptr() as *mut Self;
-        mem::forget(v);
+        let layout = Self::get_layout(height);
+        let ptr = alloc(layout) as *mut Self;
```

Comment:
This needs to check for null and call `handle_alloc_error(layout)`.

Reply:
Oh, thanks for pointing this out! I'm used to `Box::new()` always working so the possibility of a failure didn't even cross my mind.

@tobz Would you like to submit a fix for this?

Reply:
I'll submit a fix, yeah.